### DISCUSSION TITLE
Android 14

### DISCUSSION THREAD

**Discussion by sensorial7** [AndreyPavlenko/Fermata#315] [Q&A]
Hello, There is a problem with the Android 14 platform, after installing Fermata Auto and Control version 1.9.5 via Kinginstaller 1.4, there are no fermata icons in the AndroidAuto service. Everything worked on the previous version of Android 13. When can we expect the new version of Fermata dedicated to Android 14

**Comment by dm17ryk** [comment]
_On my Samsung **Galaxy S23 Ultra**, **FermataAuto** stopped working. Just I don't see it in **AndroidAuto** from now. **FermataControl** do exist in **AndroidAuto**. Tried to reinstall and reinstall with clean data, clean data of **AndroidAuto** and reinstall clean **AndroidAuto**, nothing help.
**AndroidAuto**: 10.8.634424-release
**Android**: 14_

**I resolved it by compiling it by myself and added some keys to manifests files. I did it according to:
https://developer.android.com/training/cars/media/auto**

to:
**control/src/main/AndroidManifest.xml**
added:
`<uses-sdk android:minSdkVersion="23" android:targetSdkVersion="34" />`

to:
**fermata/src/auto/res/xml/automotive_app_desc.xml**
added:
`<uses name="media" />`
 
to:
**fermata/src/main/AndroidManifest.xml**
added:
`<uses-sdk android:minSdkVersion="23" android:targetSdkVersion="34" />`

If anybody has the same issue, maybe it will help.

**Reply by dm17ryk** [reply]
Actually, maybe only this is needed:

to:
**fermata/src/auto/res/xml/automotive_app_desc.xml**
added:
`<uses name="media" />`

I see that min SDK and target SDK are set via Gradle.

**Reply by akulp** [reply]
@dmitrykok thnak you!

I managed to get Fermata Auto on my AA.

AndroidAuto: 10.8.634424-release
Android: 14
Pixel 7 Pro

But in Fermata Auto on AA I don't see Web browser/YT option, but I see them on my mobile.

I'm using Fermata for the first time, so maybe there's something else I should do?

![Paa](https://github.com/AndreyPavlenko/Fermata/assets/5618615/ddba1450-32da-4fbf-88d0-3d23dd532b56)

**Reply by Creazawolf** [reply]
i have the exact same phone and android 14. Is compiling it myself the only way to get this to work?

**Reply by lmsimonep** [reply]
Could you share your precompiled version?

**Comment by emsss78** [comment]
I have an S23 Ultra as well and I can't get Fermata to work after Android 14 update...will it be patched anytime soon?? or do I need to do something

**Comment by maripic69** [comment]
Има ли напредък по въпроса с Фермата за Андроид 14!!!

**Comment by Danymetal** [comment]
Good morning everyone! These are my latest news: I bought AAWireless as someone suggested (again thanks a lot!), put it in developer mode and everything works perfectly like in Android 13, and installed this USB hub with switches in my car, to also have the USB stick with music  https://www.amazon.it/dp/B0BNGQVJ17?psc=1&ref=ppx_yo2ov_dt_b_product_details   (my car has only one USB socket), and this make me able to turn AAWireless on or off whenever I want, because unlike the Carsifi It doesn't have a switch. The only thing I noticed, when I start the car for the first time of the day (the other times, if it doesn't take too long, it starts immediately instead), Aaw doesn't start... I found that if I open the Aaw app from my phone however, it connects right away...has anyone else had this experience?

**Comment by prw19** [comment]
> _On my Samsung **Galaxy S23 Ultra**, **FermataAuto** stopped working. Just I don't see it in **AndroidAuto** from now. **FermataControl** do exist in **AndroidAuto**. Tried to reinstall and reinstall with clean data, clean data of **AndroidAuto** and reinstall clean **AndroidAuto**, nothing help.
> **AndroidAuto**: 10.8.634424-release
> **Android**: 14_
> 
> **I resolved it by compiling it by myself and added some keys to manifests files. I did it according to:
> https://developer.android.com/training/cars/media/auto**
> 
> to:
> **control/src/main/AndroidManifest.xml**
> added:
> `<uses-sdk android:minSdkVersion="23" android:targetSdkVersion="34" />`
> 
> to:
> **fermata/src/auto/res/xml/automotive_app_desc.xml**
> added:
> `<uses name="media" />`
>  
> to:
> **fermata/src/main/AndroidManifest.xml**
> added:
> `<uses-sdk android:minSdkVersion="23" android:targetSdkVersion="34" />`
> 
> If anybody has the same issue, maybe it will help.

How you add key to files? Using Windows on computer? Or on the phone? Is there any tutorial how to do your solution?

**Reply by AndreyPavlenko** [reply]
It doesn't work.
